src/main/webapp/app/entities/course/course-score-calculation-component.ts

```diff
@@ -0,0 +1,199 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { Subscription } from 'rxjs/Subscription';
+import { HttpResponse } from '@angular/common/http';
+
+import { CourseService, CourseScoreCalculationService} from './course.service';
+import {Participation} from 'app/entities/participation';
+import {Result} from 'app/entities/result';
+import {Exercise} from 'app/entities/exercise';
+import {Course} from 'app/entities/course/course.model';
+
+@Component({
+    selector: 'jhi-course-score',
+    templateUrl: './course-score-calculation.component.html'
+})
+export class CourseScoreCalculationComponent implements OnInit, OnDestroy {
+
+    private courseId: number;
+    private courseExercises: Exercise[];
+    private subscription: Subscription;
+    private SCORE_NORMALIZATION_VALUE = 0.01;
+    course: Course;
+
+    // absolute score
+    totalScore = 0;
+    quizzesTotalScore = 0;
+    programmingExerciseTotalScore = 0;
+    modelingExerciseTotalScore = 0;
+
+    // relative score
+    totalRelativeScore = 0;
+    quizzesTotalRelativeScore = 0;
+    programmingExerciseTotalRelativeScore = 0;
+    modelingExerciseTotalRelativeScore = 0;
+
+    // max score
+    totalMaxScore = 0;
+    quizzesTotalMaxScore = 0;
+    programmingExerciseTotalMaxScore = 0;
+    modelingExerciseTotalMaxScore = 0;
+
+    constructor(
+        private courseService: CourseService,
+        private courseCalculationService: CourseScoreCalculationService,
+        private courseServer: CourseService,
+        private route: ActivatedRoute) {}
+
+    ngOnInit() {
+        this.subscription = this.route.params.subscribe(params => {
+            this.courseId = params['id'];
+        });
+
+        this.course = this.courseCalculationService.getCourse(this.courseId);
+
+        if(this.course == undefined) {
+            console.log('Blank');
```

Comment:
please remove log statements for the production use

Reply:
My bad. Removed :D